test/object-store/transaction_log_parsing.cpp

```diff
@@ -2242,3 +1988,354 @@ TEMPLATE_TEST_CASE("DeepChangeChecker", "[notifications]", ListOfObjects, ListOf
+    std::pair<TableKey, ColKey> pair_link(table->get_key(), table->get_column_key("link1"));
+    std::pair<TableKey, ColKey> pair_mixed_link(table->get_key(), table->get_column_key("mixed_link"));
+    KeyPathArray key_path_array_int = {{pair_int}};
+    KeyPathArray key_path_array_link = {{pair_link}};
+    KeyPathArray key_path_mixed_link = {{pair_mixed_link}};
+    KeyPathArray key_path_array_empty = {};
+    auto relation_updater = [&]() {
+        related_tables.clear();
+        _impl::DeepChangeChecker::find_related_tables(related_tables, *table, key_path_array_empty);
+    };
+    relation_updater();
+
+    auto set_link = [&](Obj obj, ColKey col, ObjLink link) {
+        if (col.get_type() == col_type_Mixed) {
+            obj.set(col, Mixed{link});
+        }
+        else {
+            obj.set(col, link.get_obj_key());
+        }
+    };
+
+    auto cols = table->get_column_keys();
+    SECTION("direct changes are tracked") {
+        auto info = track_changes([&] {
+            table->get_object(9).set(cols[0], 10);
+        });
+
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_empty, false)(8));
+        REQUIRE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_empty, false)(9));
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_empty, true)(8));
+        REQUIRE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_empty, true)(9));
+
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_int, false)(8));
+        REQUIRE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_int, false)(9));
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_int, true)(8));
+        REQUIRE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_int, true)(9));
+
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_link, false)(8));
+        REQUIRE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_link, false)(9));
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_link, true)(8));
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_link, true)(9));
+
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_mixed_link, false)(8));
+        REQUIRE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_mixed_link, false)(9));
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_mixed_link, true)(8));
+        REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_mixed_link, true)(9));
+    }
+
+    SECTION("changes over links are tracked") {
+        bool did_run_section = false;
+
+        {
+            size_t col_ndx = GENERATE(1, 2, 3);
```

Comment:
these tests are mostly the same as before, except that they use catch2's GENERATE to duplicate everything for each link column in the schema so that we can cover mixed as well

Reply:
Neat, I didn't know that was a thing. It's nicer than injecting a template type just for this.